socs/agents/lakeshore240/agent.py

```diff
@@ -1,309 +1,326 @@
+    )
+    pgroup.add_argument(
+        "--sampling-frequency",
+        type=float,
+        help="Sampling frequency for data acquisition",
+    )
 
     return parser
 
 
 def main(args=None):
     # Start logging
-    txaio.start_logging(level=os.environ.get("LOGLEVEL", "info"))
+    txaio.start_logging(level=os.environ.get("LOGLEVEL", "info"))  # pylint: disable=E1101
 
     parser = make_parser()
 
     # Not used anymore, but we don't it to break the agent if these args are passed
-    parser.add_argument('--fake-data', help=argparse.SUPPRESS)
-    parser.add_argument('--num-channels', help=argparse.SUPPRESS)
+    parser.add_argument("--fake-data", help=argparse.SUPPRESS)
+    parser.add_argument("--num-channels", help=argparse.SUPPRESS)
 
     # Interpret options in the context of site_config.
-    args = site_config.parse_args(agent_class='Lakeshore240Agent',
-                                  parser=parser,
-                                  args=args)
+    args = site_config.parse_args(
+        agent_class="Lakeshore240Agent", parser=parser, args=args
+    )
 
     if args.fake_data is not None:
-        warnings.warn("WARNING: the --fake-data parameter is deprecated, please "
-                      "remove from your site-config file", DeprecationWarning)
+        warnings.warn(
+            "WARNING: the --fake-data parameter is deprecated, please "
+            "remove from your site-config file",
+            DeprecationWarning,
+        )
 
     if args.num_channels is not None:
-        warnings.warn("WARNING: the --num-channels parameter is deprecated, please "
-                      "remove from your site-config file", DeprecationWarning)
-
-    # Automatically acquire data if requested (default)
-    init_params = False
-    if args.mode == 'init':
-        init_params = {'auto_acquire': False}
-    elif args.mode == 'acq':
-        init_params = {'auto_acquire': True}
+        warnings.warn(
+            "WARNING: the --num-channels parameter is deprecated, please "
+            "remove from your site-config file",
+            DeprecationWarning,
+        )
+
+    if args.mode is not None:
+        warnings.warn(
+            "WARNING: the --init-mode parameter is deprecated, please "
+            "remove from your site-config file",
+            DeprecationWarning,
+        )
 
     device_port = None
     if args.port is not None:
         device_port = args.port
     else:  # Tries to find correct USB port automatically
-
         # This exists if udev rules are setup properly for the 240s
-        if os.path.exists('/dev/{}'.format(args.serial_number)):
+        if os.path.exists("/dev/{}".format(args.serial_number)):
             device_port = "/dev/{}".format(args.serial_number)
 
-        elif os.path.exists('/dev/serial/by-id'):
-            ports = os.listdir('/dev/serial/by-id')
+        elif os.path.exists("/dev/serial/by-id"):
+            ports = os.listdir("/dev/serial/by-id")
```

Comment:
Yes... sorry I did this in the PR, but I did run this through Ruff auto-formatting.  Its very nice and I think would be a good addition to the pre-commit. Since this is a big rewrite I thought it would be ok to run on this file, but in general it can distract from actual changes, so I think we'd want to run on the whole repo before implementing the pre-commit.